<?xml version="1.0" encoding="utf-8"?>
<?xml-stylesheet type="text/xsl" href="../assets/xml/rss.xsl" media="all"?><rss version="2.0" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:atom="http://www.w3.org/2005/Atom"><channel><title>The Cloistered Monkey (Posts about python errors)</title><link>https://necromuralist.github.io/</link><description></description><atom:link href="https://necromuralist.github.io/categories/python-errors.xml" rel="self" type="application/rss+xml"></atom:link><language>en</language><lastBuildDate>Fri, 09 Mar 2018 00:08:03 GMT</lastBuildDate><generator>Nikola (getnikola.com)</generator><docs>http://blogs.law.harvard.edu/tech/rss</docs><item><title>Fatal Python Error</title><link>https://necromuralist.github.io/posts/fatal-platform-error/</link><dc:creator>Cloistered Monkey</dc:creator><description>&lt;div&gt;&lt;p&gt;I was going to make my first &lt;cite&gt;nikola&lt;/cite&gt; post in a few months but when I tride the &lt;tt class="docutils literal"&gt;nikola new_post&lt;/tt&gt; command I got the following error.&lt;/p&gt;
&lt;pre class="code bash"&gt;&lt;a name="rest_code_192ee2d188374b849735b7c4d67713f9-1"&gt;&lt;/a&gt;Could not find platform independent libraries &amp;lt;prefix&amp;gt;
&lt;a name="rest_code_192ee2d188374b849735b7c4d67713f9-2"&gt;&lt;/a&gt;Could not find platform dependent libraries &amp;lt;exec_prefix&amp;gt;
&lt;a name="rest_code_192ee2d188374b849735b7c4d67713f9-3"&gt;&lt;/a&gt;Consider setting &lt;span class="nv"&gt;$PYTHONHOME&lt;/span&gt; to &amp;lt;prefix&amp;gt;&lt;span class="o"&gt;[&lt;/span&gt;:&amp;lt;exec_prefix&amp;gt;&lt;span class="o"&gt;]&lt;/span&gt;
&lt;a name="rest_code_192ee2d188374b849735b7c4d67713f9-4"&gt;&lt;/a&gt;Fatal Python error: Py_Initialize: Unable to get the locale encoding
&lt;a name="rest_code_192ee2d188374b849735b7c4d67713f9-5"&gt;&lt;/a&gt;ImportError: No module named &lt;span class="s1"&gt;'encodings'&lt;/span&gt;
&lt;/pre&gt;&lt;p&gt;I had no idea what this meant so I tried searching the web for the error and found people saying different things about what it meant to them when they encountered it, but the one that pointed the way for me was a &lt;a class="reference external" href="https://github.com/pypa/virtualenv/issues/161"&gt;bug report for virtualenv&lt;/a&gt; where a user reported that he got this error because, it turned out, the Windows version didn't work with symlinks if the window was opened as an administrator.&lt;/p&gt;
&lt;p&gt;I'm not using Windows, but when  I changed into the directory for my &lt;cite&gt;nikola&lt;/cite&gt; &lt;cite&gt;virtualenv&lt;/cite&gt; installation, &lt;tt class="docutils literal"&gt;ls &lt;span class="pre"&gt;-l&lt;/span&gt;&lt;/tt&gt; showed that all my symbolic links were broken. I don't know how it happened... maybe something got moved, but the point of this post was to make a note for myself if I see this error again - check the sym-links for the virtualenv installation.&lt;/p&gt;&lt;/div&gt;</description><category>python errors</category><guid>https://necromuralist.github.io/posts/fatal-platform-error/</guid><pubDate>Sun, 19 Jun 2016 20:53:31 GMT</pubDate></item></channel></rss>